doc/developer/platform/ux.md

```diff
@@ -299,17 +299,63 @@ The HTTPS server exposes two APIs:
     additional control over query execution and result fetching, a la the
     [Snowflake SQL REST API].
 
-    TODO: fully describe the output format.
+    ### Output format
 
-    TODO: design support for binding parameters.
+    The output format is a JSON object with one key, `results`, whose values is
+    an array of serialized `adapter::command::SimpleResult`, representing all of
+    the statements' executions.
+
+    The serialized `adapter::command::SimpleResult` values are representing in JSON as:
+
+    `SimpleResult` value | JSON value
```

Comment:
Can this specify that there will always be exactly one of `Error` and `Ok`, and `Rows` only appears when there are rows?

Reply:
Clarified; there will only ever be one of the three in response to a statement, e.g. `INSERT...RETURNING` only returned `Rows` and the `Ok` is implicit. This is how responses currently work (1:1) and refactoring seems like something outside the scope of the initial design.

Reply:
Closing this, but the question remains open if we can get the appropriate tag value for all execute responses, e.g. `SendingRows`